### DISCUSSION TITLE
Resetting list variable produces error

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4347] [1. Help]
I finally found out the reason for the following error:

![image](https://github.com/user-attachments/assets/76d9b7eb-8ae8-4440-8de5-83d9d851f0e9)

It is, when I reset the list, I set `this.list = []`. This causes this error. However, I can't understand why, because the list never is undefined?

However, if I use `this.list.length = 0`, there is no error. Why?

**Comment by ekwoka** [comment]
Would need to see more code. That's just too small of a context.

**Comment by simonferndriger** [comment]
Well, I tried to create a reproduction pen, but unfortunately, there it works without issue: https://codepen.io/simonferndriger/pen/OJevYwm

I guess the only part of code that might help is the html in the template loop, otherwise it's for sure too much to ask in regards of the component code.

![image](https://github.com/user-attachments/assets/789ff67d-376b-4362-99b2-4ad8bbd428d9)
